accepted/2021/statics-in-interfaces/README.md

```diff
@@ -6,8 +6,1267 @@
+
+        static abstract TResult checked operator +(TSelf lhs, TOther rhs);
+    }
+
+    public interface ISubtractable<TSelf, TOther, TResult>
+        where TSelf : ISubtractable<TSelf, TOther, TResult>
+    {
+        static abstract TResult operator -(TSelf lhs, TOther rhs);
+
+        static abstract TResult checked operator -(TSelf lhs, TOther rhs);
+    }
+
+    public interface IIncrementable<TSelf>
+        where TSelf : IIncrementable<TSelf>
+    {
+        // We don't inherit IAddable as TOther isn't well-defined for incrementable types
+        // Incrementing twice is not necessarily the same as self + 2 and not all languages
+        // use incrementing for numbers, they are sometimes used by iterators for example
+
+        // We can't expose TSelf-- as some types only support one of the operations
+
+        // We don't support TResult as C# requires TSelf to be the return type
+
+        static abstract TSelf operator ++(TSelf value);
+
+        static abstract TSelf checked operator ++(TSelf value);
+    }
+
+    public interface IDecrementable<TSelf>
+        where TSelf : IDecrementable<TSelf>
+    {
+        static abstract TSelf operator --(TSelf value);
+
+        static abstract TSelf checked operator --(TSelf value);
+    }
+
+    public interface IMultipliable<TSelf, TOther, TResult>
+        where TSelf : IMultipliable<TSelf, TOther, TResult>
+    {
+        // A multiplicative identity is not exposed here for the same reasons as IAddable<TSelf, TOther>
+
+        // We can't assume TOther * TSelf is valid as not everything is commutative
+
+        // We can't expose TSelf / TOther as some types, such as matrices, don't support dividing with respect to TOther
+
+        // We can't inherit from IAddable<TSelf, TOther> as some types, such as matrices, support Matrix * scalar but not Matrix + scalar
+
+        static abstract TResult operator *(TSelf lhs, TOther rhs);
+
+        static abstract TResult checked operator *(TSelf lhs, TOther rhs);
+    }
+
+    public interface IDivisable<TSelf, TOther, TResult>
+        where TSelf : IDivisable<TSelf, TOther, TResult>
+    {
+        static abstract TResult operator /(TSelf lhs, TOther rhs);
+
+        static abstract TResult checked operator /(TSelf lhs, TOther rhs);
+    }
+
+    public interface IRemainder<TSelf, TOther, TResult>
+        where TSelf : IRemainder<TSelf, TOther, TResult>
+    {
+        // The ECMA name is op_Modulus and so the behavior isn't actually "remainder" with regards to negative values
+
+        // Likewise the name IRemainder doesn't fit with the other names, so a better one is needed
+
+        static abstract TResult operator %(TSelf lhs, TOther rhs);
+
+        static abstract TResult checked operator %(TSelf lhs, TOther rhs);
+    }
+
+    public interface INegatable<TSelf, TResult>
+        where TSelf : INegatable<TSelf, TResult>
+    {
+        // Should unary plus be on its own type?
```

Comment:
If not, this seems like a good case for supporting default implementations since (I think it's safe to say) 99% of implementations of unary `+` will just return `value` unmodified.